src/coreclr/nativeaot/Runtime/riscv64/UniversalTransition.S

```diff
@@ -85,87 +84,88 @@
+
+        PREPARE_EXTERNAL_VAR RhpIntegerTrashValues, t1
+
+        ld      t2, 0x10(t1)
+        ld      t3, 0x18(t1)
+        ld      t4, 0x20(t1)
     #endif // TRASH_SAVED_ARGUMENT_REGISTERS
 
-        addi a0, sp, DISTANCE_FROM_CHILDSP_TO_RETURN_BLOCK  // First parameter to target function
-        mv a1, t1                                         // Second parameter to target function
-        jalr t0, t1, 0                                    // Jump to the function in t1
+        addi    a0, sp, 144             // First parameter to target function is a pointer to the return block
+        mv      a1, t2                  // Second parameter to target function remains in a1
+        jalr    a2                      // Jump to the managed target address in a2
 
     ALTERNATE_ENTRY ReturnFrom\FunctionName
 
-        // Restore the result address from t2
-        mv t2, a0                                       // Move result to t2
-
-        // Restore Floating Point registers
-        fld f0, 0x100(sp)         // Restore f0
-        fld f1, 0x108(sp)         // Restore f1
-        fld f2, 0x110(sp)         // Restore f2
-        fld f3, 0x118(sp)         // Restore f3
-        fld f4, 0x120(sp)         // Restore f4
-        fld f5, 0x128(sp)         // Restore f5
-        fld f6, 0x130(sp)         // Restore f6
-        fld f7, 0x138(sp)         // Restore f7
-
-        // Restore argument registers
-        ld a0, 0x140(sp)         // Restore a0
-        ld a1, 0x148(sp)         // Restore a1
-        ld a2, 0x150(sp)         // Restore a2
-        ld a3, 0x158(sp)         // Restore a3
-        ld a4, 0x160(sp)         // Restore a4
-        ld a5, 0x168(sp)         // Restore a5
-        ld a6, 0x170(sp)         // Restore a6
-        ld a7, 0x178(sp)         // Restore a7
+        // Move the result (the target address) to a temporary register so it doesn't get overridden
+        mv      t1, a0
+
+        // Restore floating-point registers (fa0-fa7)
+        fld     fa0, 0(sp)
+        fld     fa1, 8(sp)
+        fld     fa2, 16(sp)
+        fld     fa3, 24(sp)
+        fld     fa4, 32(sp)
+        fld     fa5, 40(sp)
+        fld     fa6, 48(sp)
+        fld     fa7, 56(sp)
+
+        // Restore argument registers (a0 - a7)
+        ld      a0, 64(sp)
+        ld      a1, 72(sp)
+        ld      a2, 80(sp)
+        ld      a3, 88(sp)
+        ld      a4, 96(sp)
+        ld      a5, 104(sp)
+        ld      a6, 112(sp)
+        ld      a7, 120(sp)
 
         // Restore FP and RA registers, and free the allocated stack block
-        EPILOG_RESTORE_REG_PAIR_INDEXED  fp, ra, STACK_SIZE
+        ld      ra, 264(sp)             // Restore return address
+        ld      fp, 256(sp)             // Restore frame pointer
+        addi    sp, sp, 272             // Adjust stack pointer back to original
 
-        // Tailcall to the target address in t2
-        jalr t2, 0
+        // Perform the return to caller (via jump)
```

Comment:
Is this a dump or is this a return? If it is a return, should it use `ret` instruction?

Reply:
It is jumping to address in specific registers just like arm64.